airport-challenge-solution.md

```diff
@@ -0,0 +1,158 @@
+|         |                             | getLandedPlaneCount()       | @Integer |
+|         |                             | takeOffPlane(@Plane)        | @Void    |
+|         |                             | isPlanePresent(@Plane)      | @Boolean |
+| Plane   | id @String                  | getId()                     | @String  |
+
+## Tests - 7&8
+
+7 landPlane Plane(id) and check if isPlanePresent (T/F)
+8 in the same way - takeOffPlane Plane(id) and check if isPlanePresent (T/F)
+
+## Part 5 User Story
+
+As an air traffic controller
+To avoid confusion
+I want to prevent asking the airport to let planes take-off which are not at the airport, or land a plane that's already landed
+
+## Part 5 Domain Model (not added to) - already fine from above
+
+| Objects | Properties                  | Messages                    | Outputs  |
+| ------- | --------------------------- | --------------------------- | -------- | ----------- |
+| Airport | landedPlanes @Array[@Plane] | landPlane(@Plane)           | @Void    |
+|         | landedCapacity @Integer     | isAirportFull()             | @Boolean | did not use |
+|         |                             | getLandedCapacity()         | @Integer |
+|         |                             | newLandedCapacity(@Integer) | @Void    |
+|         |                             | getLandedPlaneCount()       | @Integer |
+|         |                             | takeOffPlane(@Plane)        | @Void    |
+|         |                             | isPlanePresent(@Plane)      | @Boolean |
+| Plane   | id @String                  | getId()                     | @String  |
+
+## Tests 9&10
+
+9. try to take off a plane(id) which is not in the airport (landedPlanes)
+10. try land a second instance of a plane(id) that is already in landedPlanes
+    (need to add a Boolean return to landPlane function?, yes but did not end up using)
+
+## Testing Edge Cases
+
+Have already shown it is not possible to land something that is not a plane - a string and a value, in test 3.
+Have also shown that a plane that is already landed can not land a second time, in test 10.
+
+## Test 11 - Test to check that a plane that has taken off cannot take off again.
+
+Same approach test as previous (test 10), did use a Boolean return.
+
+## Test 12 - Test to check that where there are integers - a -ve value or a float cannot be accepted.
+
+defaultAirportCapacity cannot be a decimal (float) or negative value, only a positive integer.
+
+#### Extended Acceptance Criteria
+
+```
+As an air traffic controller
+To ensure safety
+I want to prevent takeoff when weather is stormy //not attempted
+
+As an air traffic controller
+To ensure safety
+I want to prevent landing when weather is stormy //not attempted
+
+As an air traffic controller
+To count planes easily
+Planes that have landed must be at an airport //this is covered within the Airport Class, landedPlanes array.
+```
```

Comment:
Read ahead to extended criteria to understand if I am attempting them within my earlier tests.